Bottom: 159826520d8722d3a77119f6b7a139e3f2d772c3
Top:    23573954eaa537022522f6fb586e9255886d08a5
Author: Jason Wang <jasowang@redhat.com>
Date:   2015-09-22 17:31:14 +0800

Refresh of net-merge-qemu_deliver_packet

---

diff --git a/net/net.c b/net/net.c
index ca35b27..38c92ab 100644
--- a/net/net.c
+++ b/net/net.c
@@ -689,10 +689,17 @@ ssize_t qemu_send_packet_raw(NetClientState *nc, const uint8_t *buf, int size)
 static ssize_t nc_sendv_compat(NetClientState *nc, const struct iovec *iov,
                                int iovcnt, unsigned flags)
 {
-    uint8_t buffer[NET_BUFSIZE];
+    uint8_t buf[NET_BUFSIZE];
+    uint8_t *buffer;
     size_t offset;
 
-    offset = iov_to_buf(iov, iovcnt, 0, buffer, sizeof(buffer));
+    if (iovcnt == 1) {
+        buffer = iov[0].iov_base;
+        offset = iov[0].iov_len;
+    } else {
+        buffer = buf;
+        offset = iov_to_buf(iov, iovcnt, 0, buffer, sizeof(buffer));
+    }
 
     if (flags & QEMU_NET_PACKET_FLAG_RAW && nc->info->receive_raw) {
         return nc->info->receive_raw(nc, buffer, offset);
